types/coin.go

```diff
@@ -30,6 +31,9 @@ func NewCoin(denom string, amount Int) Coin {
 	if amount.LT(ZeroInt()) {
 		panic(fmt.Sprintf("negative coin amount: %v\n", amount))
 	}
+	if len(strings.TrimSpace(denom)) == 0 {
```

Comment:
@cwgoes @alexanderbez Happy to revert this commit if I'm mistaken - why do we allow an empty denom coin to pass validation?

Reply:
We don't from a UX perspective afaik, as `ParseCoins` would fail. Ideally, this should be a separate PR imho.